torch_xla/csrc/tensor_util.cpp

```diff
@@ -653,6 +653,16 @@ torch::lazy::BackendDataPtr TensorToXlaData(
     const at::Tensor& tensor, const xla::Shape& shape,
     const torch::lazy::BackendDevice& device) {
   XLA_TIMED("TensorToData");
+  if (device.toString() == "SPMD:0") {
+    // When SPMD is enabled, we want to delay the data transfer for XLA
+    // tensors until the data is sharded. So, we skip the data transfer
+    // here and simply return a placeholder for the backend data ptr.
+    // Data will only be transferred via CreateTensorsData, when users
+    // call the mark_sharding API.
+    return WrapXlaData(
+        xla::ComputationClient::Get()->CreateDataPlaceholder("SPMD:0", shape));
```

Comment:
nit. Use `device` instead of "SPMD:0" and also check for device `type()` is `XlaDeviceType::SPMD`.

Reply:
Done